src/lib/libscf_solver/hf.cc

```diff
@@ -405,12 +405,12 @@ void HF::integrals()
           #ifdef HAVE_JK_FACTORY
             //DGAS is adding to the ghetto, this Python -> C++ -> C -> C++ -> back to C is FUBAR
             boost::shared_ptr<Molecule> other_legacy = Process::environment.legacy_molecule();
-            Process::environment.set_legacy_wavefunction(molecule_);
+            //Process::environment.set_legacy_wavefunction(Process::environment.legacy_wavefunction());
+            Process::environment.set_legacy_molecule(other_legacy);
             if(options_.get_bool("SOSCF"))
                 jk_ = boost::shared_ptr<JK>(new GTFockJK(basisset_,2,false));
             else
                 jk_ = boost::shared_ptr<JK>(new GTFockJK(basisset_,2,true));
-            Process::environment.set_legacy_molecule(other_legacy);
           #else
```

Comment:
Right, you have to set a legacy molecule, but it should be the HF.cc molecule. With the code you had you were grading the current legacy molecule and just setting it again, which effectively does nothing.